<h1 id="compseq">compseq</h1>

<p>Calculate the composition of unique words in sequences.
More information: <a href="https://www.bioinformatics.nl/cgi-bin/emboss/help/compseq/">https://www.bioinformatics.nl/cgi-bin/emboss/help/compseq/</a>.</p>

<ul>
  <li>Count observed frequencies of words in a FASTA file, providing parameter values with interactive prompt:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/file.fasta</span></p>

<ul>
  <li>Count observed frequencies of amino acid pairs from a FASTA file, save output to a text file:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/input_protein.fasta</span><code class="language-plaintext highlighter-rouge"> -word 2 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file.comp</span></p>

<ul>
  <li>Count observed frequencies of hexanucleotides from a FASTA file, save output to a text file and ignore zero counts:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/input_dna.fasta</span><code class="language-plaintext highlighter-rouge"> -word 6 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file.comp</span><code class="language-plaintext highlighter-rouge"> -nozero</code></p>

<ul>
  <li>Count observed frequencies of codons in a particular reading frame; ignoring any overlapping counts (i.e. move window across by word-length 3):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq -sequence </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/input_rna.fasta</span><code class="language-plaintext highlighter-rouge"> -word 3 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file.comp</span><code class="language-plaintext highlighter-rouge"> -nozero -frame </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">1</span></p>

<ul>
  <li>Count observed frequencies of codons frame-shifted by 3 positions; ignoring any overlapping counts (should report all codons except the first one):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq -sequence </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/input_rna.fasta</span><code class="language-plaintext highlighter-rouge"> -word 3 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file.comp</span><code class="language-plaintext highlighter-rouge"> -nozero -frame 3</code></p>

<ul>
  <li>Count amino acid triplets in a FASTA file and compare to a previous run of <code class="language-plaintext highlighter-rouge">compseq</code> to calculate expected and normalised frequency values:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq -sequence </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/human_proteome.fasta</span><code class="language-plaintext highlighter-rouge"> -word 3 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file1.comp</span><code class="language-plaintext highlighter-rouge"> -nozero -infile </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file2.comp</span></p>

<ul>
  <li>Approximate the above command without a previously prepared file, by calculating expected frequencies using the single base/residue frequencies in the supplied input sequence(s):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq -sequence </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/human_proteome.fasta</span><code class="language-plaintext highlighter-rouge"> -word 3 </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">path/to/output_file.comp</span><code class="language-plaintext highlighter-rouge"> -nozero -calcfreq</code></p>

<ul>
  <li>Display help (use <code class="language-plaintext highlighter-rouge">-help -verbose</code> for more information on associated and general qualifiers):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">compseq -help</code></p>
