<h1 id="samtools">samtools</h1>

<p>Tools for handling high-throughput sequencing (genomics) data.
Used for reading/writing/editing/indexing/viewing of data in SAM/BAM/CRAM format.
More information: <a href="https://www.htslib.org">https://www.htslib.org</a>.</p>

<ul>
  <li>Convert a SAM input file to BAM stream and save to file:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools view -S -b </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">input.sam</span><code class="language-plaintext highlighter-rouge"> &gt; </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">output.bam</span></p>

<ul>
  <li>Take input from <code class="language-plaintext highlighter-rouge">stdin</code> (-) and print the SAM header and any reads overlapping a specific region to <code class="language-plaintext highlighter-rouge">stdout</code>:</li>
</ul>

<p><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">other_command</span><code class="language-plaintext highlighter-rouge"> | samtools view -h - chromosome:start-end</code></p>

<ul>
  <li>Sort file and save to BAM (the output format is automatically determined from the output file's extension):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools sort </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">input</span><code class="language-plaintext highlighter-rouge"> -o </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">output.bam</span></p>

<ul>
  <li>Index a sorted BAM file (creates <code class="language-plaintext highlighter-rouge">sorted_input.bam.bai</code>):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools index </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">sorted_input.bam</span></p>

<ul>
  <li>Print alignment statistics about a file:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools flagstat </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">sorted_input</span></p>

<ul>
  <li>Count alignments to each index (chromosome/contig):</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools idxstats </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">sorted_indexed_input</span></p>

<ul>
  <li>Merge multiple files:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools merge </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">output</span><code class="language-plaintext highlighter-rouge"> </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">input1 input2 â€¦</span></p>

<ul>
  <li>Split input file according to read groups:</li>
</ul>

<p><code class="language-plaintext highlighter-rouge">samtools split </code><span class="tldr-var badge badge-pill bg-dark-lm bg-white-dm text-white-lm text-dark-dm font-weight-bold">merged_input</span></p>
